rfcs/020-automated-release-process.md

```diff
@@ -0,0 +1,84 @@
+- Feature Name: Automated Release Process
+- Start Date: 2020-12-19
+- RFC PR: [amundsen-io/rfcs#20](https://github.com/amundsen-io/rfcs/pull/20)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Automated Release Process
+
+## Summary
+
+This RFC proposes to automate release with github actions.
+
+## Motivation
+
+Currently release process requires manual release creation and changelog rendering (by going through list of commits).
+
+The motivation behind this RFC is to automate, unify and simplify release process.
+
+## Guide-level Explanation (aka Product Details)
+
+Having enforced naming convention for MRs we can leverage this to automatically categorize MRs and render them in
+changelog. By using automated github actions we can remove manual step from release process (releasing, rendering changelog)
+and replace it with three operations:
+
+- creating a git tag triggers a release
+- periodical releases every month
+- pre-releases with each change added to master
```

Comment:
my feeling is that'd be too granular (as explained in another comment)

Reply:
My thinking here is to cover this use case:

1. User reaches out about a bug
2. We fix it, merge it into master
3. We need to trigger a new release and ask user to update

Instead, we would just fix it and get it into master, asking the user to use the prerelease to get the latest

If all is automatic for us, I don't see any problem. Another thing is if we can't do it automatically, that would be tedious!